test/framework/src/main/java/org/elasticsearch/test/rest/yaml/section/DoSection.java

```diff
@@ -62,13 +63,16 @@
  *      headers:
  *          Authorization: Basic user:pass
  *          Content-Type: application/json
- *      warnings:
+ *      warnings|warnings_regex:
  *          - Stuff is deprecated, yo
  *          - Don't use deprecated stuff
  *          - Please, stop. It hurts.
- *      allowed_warnings:
+ *          - The non _regex version exact matches against the warning text and no need to worry about escaped quotes or backslashes
+ *          - The _regex version matches against the raw value of the warning text which may include backlashes and quotes escaped
+ *      allowed_warnings|allowed_warnings_regex:
```

Comment:
I might break out the regex ones just to show what a regex will look like. I agree on preferring the non-regex one. Its nice not to have to think about what punctuation means.